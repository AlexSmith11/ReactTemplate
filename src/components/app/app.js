import React, { Component } from 'react';
import { connect } from 'react-redux';

import style from './app.scss';

// Note: We export both the standard component AND the connected component
//       so we can run tests on the component without the connection
//       getting in the way.
export class App extends Component {
  constructor(props) {
    super(props);
    this.bindMethods();
  }

  bindMethods() {
    this.onClick = this.onClick.bind(this);
  }

  render() {
    const appProps = {
      className: 'app',
      onClick: this.onClick
    };

    return (
      <div {...appProps}>
        <span>Click to make number go up: {this.props.clicks}</span>
      </div>
    );
  }

  onClick(e) {
    this.props.dispatch({
      type: 'SET_CLICKS',
      value: this.props.clicks + 1
    });
  }
}

function mapStateToProps(state) {
  return { clicks: state.clicks.value };
}

export default connect(mapStateToProps)(App);
