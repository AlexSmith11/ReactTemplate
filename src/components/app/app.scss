@import '../../theme.scss';

.app {
  background-color: $primary-color;
  padding: 80px;
  display: inline-block;
  color: $secondary-color;
  font: 2em 'Open Sans', sans-serif;
}
