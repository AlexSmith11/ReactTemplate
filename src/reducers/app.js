import { combineReducers } from 'redux';
import clicks from './clicks';

const appReducer = combineReducers({
  clicks
});

export default appReducer;
