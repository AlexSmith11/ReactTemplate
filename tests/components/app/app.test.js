import React from 'react';
import renderer from 'react-test-renderer';

import { App } from '../../../src/components/app/app.js';

test('Component renders, matching snapshot', () => {
  const tree = renderer.create(<App />).toJSON();

  expect(tree).toMatchSnapshot();
});
