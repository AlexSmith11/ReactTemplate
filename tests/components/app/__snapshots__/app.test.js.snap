// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Renders 1`] = `
<div
  className="app"
  onClick={[Function]}
>
  <span>
    Click to make number go up: 
  </span>
</div>
`;

exports[`Component renders 1`] = `
<div
  className="app"
  onClick={[Function]}
>
  <span>
    Click to make number go up: 
  </span>
</div>
`;

exports[`Component renders, matching snapshot 1`] = `
<div
  className="app"
  onClick={[Function]}
>
  <span>
    Click to make number go up: 
  </span>
</div>
`;
